src/main/java/org/folio/circulation/domain/policy/library/EndOfCurrentHoursStrategy.java

```diff
@@ -0,0 +1,42 @@
+package org.folio.circulation.domain.policy.library;
+
+import org.folio.circulation.support.HttpResult;
+import org.folio.circulation.support.ValidationErrorFailure;
+import org.folio.circulation.support.http.server.ValidationError;
+import org.joda.time.DateTime;
+import org.joda.time.DateTimeZone;
+
+import java.util.Collections;
+
+public class EndOfCurrentHoursStrategy extends ShortTermLoansBaseStrategy {
+
+  private final DateTime currentTime;
+
+  public EndOfCurrentHoursStrategy(DateTime currentTime, DateTimeZone zone) {
+    super(zone);
+    this.currentTime = currentTime;
+  }
+
+  @Override
+  protected HttpResult<DateTime> calculateIfClosed(LibraryTimetable libraryTimetable, LibraryInterval requestedInterval) {
+    LibraryInterval currentTimeInterval = libraryTimetable.findInterval(currentTime);
+    if (currentTimeInterval == null) {
+      return HttpResult.failed(errorForAbsentCurrentInterval());
+    }
+    if (!currentTimeInterval.isOpen()) {
+      return HttpResult.failed(errorForClosedCurrentInterval());
+    }
+    return HttpResult.succeeded(currentTimeInterval.getEndTime());
+  }
+
+
+  private ValidationErrorFailure errorForAbsentCurrentInterval() {
+    String message = "Unable to find current service point hours";
+    return new ValidationErrorFailure(new ValidationError(message, Collections.emptyMap()));
+  }
+
+  private ValidationErrorFailure errorForClosedCurrentInterval() {
+    String message = "Current service point hours are closed";
```

Comment:
No it wasn't. Now it is more like stub for the case when applying 'move to the end of current service pint hours' strategy is impossible. I consider it to be defined later.
Suggested temporary message from my side is: Applying 'Move to the end of current service point hours' strategy failed as library is defined to be closed now.

Reply:
If it is a stub, that's cool. Is this expected to be shown in the UI at some point? If so, we should probably get to the final version sooner rather than later in case UI unit tests refer to this message.

Reply:
Yes, this error message is expected to be shown on UI, but there is no requirements about that for now. As far as I understood, there might be some logic depending on what is the reason of failure later.